core/mm/core_mmu.c

```diff
@@ -373,8 +419,15 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 	 */
 
 #ifdef CFG_SECURE_DATA_PATH
-	for (pmem = phys_sdp_mem_begin; pmem < phys_sdp_mem_end; pmem++)
-		carve_out_phys_mem(&m, &num_elems, pmem->addr, pmem->size);
+	if (dtb_get_sdp_region()) {
+		DMSG("region base %p %zu", (void *)sec_sdp.paddr, sec_sdp.size);
+
+		carve_out_phys_mem(&m, &num_elems, sec_sdp.paddr, sec_sdp.size);
+	} else {
+		for (pmem = phys_sdp_mem_begin; pmem < phys_sdp_mem_end; pmem++)
+			carve_out_phys_mem(&m, &num_elems, pmem->addr,
+					   pmem->size);
+	}
```

Comment:
Should we report an error (trace or even panic) when a platform defines SDP memory areas both in DT and in its build config switches? I don't think one should take precedence, it would rather look like a platform misconfiguration that should be reported in some way.

Reply:
in fact, the current implementation is looking for a SDP DTB definition in any, by default. 
But it will fallback to the build config switches otherwise, which allow usage of TEE_SDP_TEST_MEM_BASE/TEE_SDP_TEST_MEM_SIZE.

Reply:
Maybe the implementation should support both: SDP mem pools defined from embedded DT and those defined with `register_sdp_mem()`.

Reply:
yes it's actually the current behavior.
for now the SDP mem pools defined from embedded DT is used if it exist, otherwise the register_sdp_mem() from build config switches is taken.
do you mean that we could define multiple SDP area from DT and/or from build config ?

Reply:
Optee_os current implementation allows several occurrences of `register_sdp_mem()`.
The DT binding you propose also allows several heap pools to be defined.
I guess optee_os could take all of them into account to forge a list of SDP areas.

Reply:
I would see one use case to such a list of SDP areas, it could allow SDP memory separation for several Trusted Application.
However have you another use case for supporting multiple SDP areas ?

Reply:
Flexibility.
If a platform defines SDP pools from both embedded DTB and from built-in platform implementation, there no reason to exclude some.

Reply:
done here 804bd35faadac5521310fa07e32d693e2c59814c